test-five/infrastructure/lib/infrastructure-stack.ts

```diff
@@ -0,0 +1,196 @@
+  aws_cloudwatch_actions as cw_actions,
+  aws_sns as sns,
+  aws_sns_subscriptions as subscriptions,
+  CfnOutput,
+  CfnParameter,
+  RemovalPolicy
+} from 'aws-cdk-lib';
+
+// Setting the domain name.
+const DOMAIN_NAME = process.env.DOMAIN_NAME!;
+
+/**
+ * Represents the CloudFormation stack for the TestFive application.
+ * @class
+ * @extends cdk.Stack
+ */
+export class TestFiveStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    /**
+     * 👉 Environment Variable:
+     */
+
+    const envParameter = new CfnParameter(this, "Environment", {
+      type: "String",
+      description: "The app environment, e.g. prod",
+      default: "local"
+    });
+
+    const snsRecipientParameter = new CfnParameter(this, "SnsRecipient", {
+      type: "String",
+      description: "Email recipient for alarm sns",
+    });
+
+    /**
+     * 👉 Stack Definition:
+     */
+
+    // Create an S3 bucket for the React app.
+    const reactAppBucket = new s3.Bucket(this, "ReactAppBucket", {
+      bucketName: `moshtix-fronted-${envParameter.valueAsString}`,
+      publicReadAccess: false,
+      removalPolicy: RemovalPolicy.DESTROY,
+      websiteIndexDocument: "index.html",
+      autoDeleteObjects: true,
+      // Needed in some scenarios, need more investigation.
+      // blockPublicAccess: {
+      //   blockPublicAcls: false,
+      //   blockPublicPolicy: false,
+      //   ignorePublicAcls: false,
+      //   restrictPublicBuckets: false,
+      // } as s3.BlockPublicAccess
+    });
+
+    // Create an Origin Access Identity (OAI) for CloudFront.
+    const cloudFrontOAI = new cloudfront.OriginAccessIdentity(this, 'OAI');
+    reactAppBucket.grantRead(cloudFrontOAI.grantPrincipal);
+
+    // Look up the hosted zone in Route 53 using the provided domain name.
+    const zone = route53.HostedZone.fromLookup(this, 'HostedZone', {
+      domainName: DOMAIN_NAME
+    });
+
+    // Create a DNS-validated certificate for the domain.
+    const siteCertificate = new acm.DnsValidatedCertificate(this, 'Certificate', {
+      domainName: DOMAIN_NAME,
+      subjectAlternativeNames: ['*.' + DOMAIN_NAME],
+      hostedZone: zone,
+      // The certificate must be issued in the us-east-1 (N. Virginia) region for it to be used with CloudFront.
+      region: 'us-east-1'
+    });
+
+    // Create a CloudFront distribution for the React app.
+    const cloudFrontWebDistribution = new cloudfront.CloudFrontWebDistribution(this, "CDKCRAStaticDistribution", {
+      originConfigs: [
+        {
+          s3OriginSource: {
+            s3BucketSource: reactAppBucket
+          },
+          behaviors: [{ isDefaultBehavior: true }]
+        }
+      ],
+      viewerCertificate: cloudfront.ViewerCertificate.fromAcmCertificate(siteCertificate, {
+        aliases: [DOMAIN_NAME, `www.${DOMAIN_NAME}`],
+        securityPolicy: cloudfront.SecurityPolicyProtocol.TLS_V1_2_2021,
```

Comment:
+1 for 1.2

Reply:
The `TLS_V1_2_2021` protocol is the recommended default protocol in AWS CloudFront, and this choice makes sense for several reasons: security, compatibility, performance, regulatory compliance, and future-proofing.